kpc/tests/test_views.py

```diff
@@ -0,0 +1,130 @@
+from django.test import TestCase, RequestFactory, Client
+from django.contrib.auth.models import Permission, AnonymousUser
+from django.contrib.contenttypes.models import ContentType
+from django.conf import settings
+from django.urls import reverse
+from model_mommy import mommy
+from kpc.views import licensee_contacts, CertificateRegisterView
+from kpc.models import Certificate
+from django.core.exceptions import PermissionDenied
+import json
+
+
+class LicenseeContactsTests(TestCase):
+
+    def setUp(self):
+        self.user = mommy.make(settings.AUTH_USER_MODEL)
+        self.factory = RequestFactory()
+        user_ct = ContentType.objects.get(app_label='accounts', model='profile')
+        self.p, _ = Permission.objects.get_or_create(content_type=user_ct, codename='can_get_licensee_contacts', name="can_get_licensee_contacts")
+        self.request = self.factory.get('')
+        self.request.user = self.user
+
+    def test_not_accessible_without_perm(self):
+        """Cannot access without perm"""
+        with self.assertRaises(PermissionDenied):
+            licensee_contacts(self.request)
+
+    def test_only_get_allowed(self):
+        """
+        405 response if request is not a GET
+        """
+        self.user.user_permissions.add(self.p)
+        for method in ['post', 'put', 'delete', 'head', 'options', 'trace']:
+            request = getattr(self.factory, method)('')
+            request.user = self.user
+            response = licensee_contacts(request)
+            self.assertEqual(response.status_code, 405)
+
+    def test_returns_empty_list_if_no_licensee(self):
+        """Return empty json list if no licensee provided"""
+        licensee = mommy.make('kpc.Licensee')
+        self.user.user_permissions.add(self.p)
+        self.user.profile.licensees.add(licensee)
+        response = licensee_contacts(self.request)
+        self.assertEqual(response.content, b'[]')
+
+    def test_returns_user_list_if_licensee(self):
+        """
+        Return json list of users associated
+        to provided licensee
+        """
+        licensee = mommy.make('kpc.Licensee')
+        self.user.user_permissions.add(self.p)
+        self.user.profile.licensees.add(licensee)
+        request = self.factory.get('', {'licensee': licensee.id})
+        request.user = self.user
+        response = licensee_contacts(request)
+        self.assertEqual(json.loads(response.content),
+                         [{'id': self.user.id, 'name': self.user.get_full_name()}])
```

Comment:
If we are using Django's `SimpleTestCase` over py.test, it might be a little less burdensome to use some of its more semantic assertions, like [assertJSONEqual](https://docs.djangoproject.com/en/2.0/topics/testing/tools/#django.test.SimpleTestCase.assertJSONEqual).